tdrs-backend/tdpservice/email/tasks.py

```diff
@@ -69,3 +73,83 @@ def email_admin_num_access_requests():
                                    subject,
                                    email_context,
                                    )
+
+
+@shared_task
+def send_data_submission_reminder(due_date, reporting_period, fiscal_quarter):
+    """Send all Data Analysts a reminder to submit if they have not already."""
+    def get_fiscal_year(calendar_year, fiscal_quarter):
+        return calendar_year - 1 if fiscal_quarter == 'Q1' else calendar_year
+
+    now = datetime.now()
+    fiscal_year = get_fiscal_year(now.year, fiscal_quarter)
+
+    all_locations = STT.objects.all()
+
+    reminder_locations = []
+    year_quarter_files = DataFile.objects.all().filter(
+        year=fiscal_year,
+        quarter=fiscal_quarter,
+    )
+
+    for loc in all_locations:
+        submitted_sections = year_quarter_files.filter(stt=loc).values_list('section', flat=True).distinct()
+        required_sections = loc.filenames.keys()
+
+        submitted_all_sections = True
+        for s in required_sections:
+            if s not in submitted_sections:
+                submitted_all_sections = False
+
+        if not submitted_all_sections:
+            reminder_locations.append(loc)
+
+    template_path = EmailType.UPCOMING_SUBMISSION_DEADLINE.value
+    text_message = f'Your datafiles are due by {due_date}.'
+
+    for loc in reminder_locations:
+        tanf_ssp_label = 'TANF and SSP' if loc.ssp else 'TANF'
+        subject = f'Action Requested: Please submit your {tanf_ssp_label} data files'
+
+        recipients = User.objects.filter(
```

Comment:
you can do similar more efficient query and push the query out of for loop, by doing a query on all reminder_locations outside the loop.

Reply:
i took a similar approach but not exactly what you suggested